README.md

```diff
@@ -38,125 +32,157 @@ deleteBlankLines path = do
 ```
 
-This function expects you to provide a handler, which should call a user-supplied error callback or success callback with the result of the asynchronous computation.
+This function expects you to provide a handler, which should call the
+supplied callback with the result of the asynchronous computation.
 
-For example, let's say we have an AJAX request function that expects a callback:
+You should also return `Canceler`, which is just a cleanup effect. Since
+`Aff` threads may be killed, all asynchronous operations should provide a
+mechanism for unscheduling it.
+
+*purescript-aff* also provides functions for easily binding FFI definitions in
+`Control.Monad.Aff.Compat`.
 
 ```javascript
-exports.ajaxGet = function(callback) { // accepts a callback
-  return function(request) { // and a request
-    return function() { // returns an effect
-      doNativeRequest(request, function(response) {
-        callback(response)(); // callback itself returns an effect
-      });
-    }
-  }
-}
+exports._ajaxGet = function (request) { // accepts a request
+  return function (onError, onSuccess) { // and callbacks
+    var req = doNativeRequest(request, function (err, response) { // make the request
+      if (err != null) {
+        onError(err); // invoke the error callback in case of an error
+      } else {
+        onSuccess(response); // invoke the success callback with the reponse
+      }
+    });
+
+    // Return a canceler, which is just another Aff effect.
+    return function (cancelError) {
+      return function (cancelerError, cancelerSuccess) {
+        req.cancel(); // cancel the request
+        cancelerSuccess(); // invoke the success callback for the canceler
+      };
+    };
+  };
+};
 ```
 
 ```purescript
-foreign import ajaxGet :: forall e. (Response -> Eff e Unit) -> Request -> Eff e Unit
+foreign import _ajaxGet :: forall eff. Request -> EffFnAff (ajax :: AJAX | eff) Response
 ```
 
 We can wrap this into an asynchronous computation like so:
 
 ```purescript
-ajaxGet' :: forall e. Request -> Aff e Response
-ajaxGet' req = makeAff (\error success -> ajaxGet success req)
+ajaxGet :: forall eff. Request -> Aff (ajax :: AJAX | eff) Response
+ajaxGet = fromEffFnAff <<< _ajaxGet
 ```
 
-This eliminates callback hell and allows us to write code simply using `do` notation:
+This eliminates callback hell and allows us to write code simply using `do`
+notation:
 
 ```purescript
-do response <- ajaxGet' req
-   liftEff $ log response.body
+example = do
+  response <- ajaxGet req
+  log response.body
 ```
 
 ## Eff
 
-All purely synchronous computations (`Eff`) can be lifted to asynchronous computations with `liftEff` defined in `Control.Monad.Eff.Class` (see [here](https://github.com/purescript/purescript-eff)).
+All purely synchronous computations (`Eff`) can be lifted to asynchronous
+computations with `liftEff` defined in `Control.Monad.Eff.Class` (see
+[here](https://github.com/purescript/purescript-eff)).
```

Comment:
May as well move the link ref onto the class name (rather than linking with a **here**) while we're doing these updates!